pkg/apis/complianceoperator/v1alpha1/complianceremediation_types.go

```diff
@@ -0,0 +1,92 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type RemediationApplicationState string
+
+const (
+	RemediationNotSelected RemediationApplicationState = "NotSelected"
+	RemediationApplied     RemediationApplicationState = "Applied"
```

Comment:
Do we need an in-between state? Such as  "Applying"?

Reply:
Initially I had this state. But then I ended up not using it, because either the remediation is part of the resulting MC and then it's applied or it's not and then it's not selected.

The only case for the mid-state would be a remediation that is marked for application (.spec.apply=true) but for some reason the MC couldn't have been created. But in that case I think the current state is OK, because you see that the spec (desired) says apply, but the status (current) says Not Selected.

btw while typing them both in a single sentence I just realized the names are odd, we should probably have either Applied/NotApplied or Selected/NotSelected..I suck at naming things.

Reply:
got it. It's fine like this then.